backend/coins/eth/account.go

```diff
@@ -207,6 +212,7 @@ func (account *Account) Initialize() error {
 }
 
 func (account *Account) poll(initDone func()) {
+	defer account.quitChanLock.RLock()()
```

Comment:
is it ok to RLock once in the start or do we need to do it for both selects ?

Reply:
should do it for every access to .quitChan. easiest to add a helper function for that which acquires the lock and returns the chan. if you do it like now at the top, it looks like a deadlock? the read-lock is never released, and Close() needs to acquire the lock to be able to close the channel.

Reply:
I thought this change would be straight forward but writing it out gave me a bunch of problems. 1. the polling gets extremely messy with re-locking/unlocking everywhere, a possible fix could be [this](https://gist.github.com/NicolaLS/c85c0eca8afced7dd8b26b88c5c28fb1) And 2. Either I am missing something or it is not really possible to use a RWMutex for this because of the `account.quitChan = nil` basically there is no guarantee that the polling will stop (read the channel) in the short time were it is unlocked after closing:
```
	quitChan, unlock := account.quitChan.WChan()
	close(*quitChan)
	unlock()
	quitChan, _ = account.quitChan.WChan()
	*quitChan = nil
	unlock()
``` 
I might be missing something here but maybe it's best to refactor the account closed checking to not use the channel that is used to quit the polling `account.quitChan == nil` should not be used to check that @benma ?

Reply:
note: I created a ```
type quitChan struct {
	mutex   sync.RWMutex
	channel chan struct{}
}
``` with a method to lock it and return the channel and the unlock fn, but I did not push this yet as I can't get the tests to pass..

Reply:
I meant helper method that looks sth like this:

```
func (account *Account) getQuichChan() <-chan struct{} {
    defer account.quitChanLock.RLock()()
    return account.quitChan
}
```

then in select:

`case <-account.getQuitChan():` etc.

Reply:
I changed it like that, PTAL. I think this is not correct though the channel gets RUnlocked as soon as the method returns. Because it is being locked swiftly it might help with synchronization but the channel is **not**  locked when we read it later (?).